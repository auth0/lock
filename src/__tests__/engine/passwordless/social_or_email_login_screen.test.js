import React from 'react';

import { expectComponent, extractPropsFromWrapper, mockComponent } from 'testUtils';

jest.mock('connection/enterprise');
jest.mock('core/index');

jest.mock('field/social/social_buttons_pane', () => mockComponent('social_buttons_pane'));
// jest.mock('core/error_screen', () => mockComponent('error_screen'));
// jest.mock('core/loading_screen', () => mockComponent('loading_screen'));
jest.mock('field/email/email_pane', () => mockComponent('email_pane'));
// jest.mock('field/captcha/captcha_pane', () => mockComponent('captcha_pane'));
jest.mock('core/pane_separator', () => mockComponent('pane_separator'));
jest.mock('connection/database/sign_up_terms', () => mockComponent('sign_up_terms'));
jest.mock('connection/passwordless/index', () => ({
  isEmail: jest.fn()
}));

const getComponent = () => {
  const SocialOrEmailScreen = require('engine/passwordless/social_or_email_login_screen').default;
  const screen = new SocialOrEmailScreen();
  return screen.render();
};

describe('email passwordless', () => {
  beforeEach(() => {
    jest.resetModules();

    jest.mock('connection/database/index', () => ({
      hasScreen: () => false
    }));

    jest.mock('connection/database/actions', () => ({
      cancelMFALogin: jest.fn(),
      logIn: jest.fn()
    }));

    jest.mock('core/signed_in_confirmation', () => ({
      renderSignedInConfirmation: jest.fn()
    }));

    jest.mock('connection/enterprise', () => ({
      isHRDEmailValid: jest.fn(() => false)
    }));

    jest.mock('core/index', () => ({
      hasSomeConnections: jest.fn(() => true)
    }));
  });

  const defaultProps = {
    i18n: {
      str: (...keys) => keys.join(','),
      html: (...keys) => keys.join(',')
    },
    model: 'model'
  };

  it('renders correctly', () => {
    const Component = getComponent();

    expectComponent(<Component {...defaultProps} />).toMatchSnapshot();
  });
});
