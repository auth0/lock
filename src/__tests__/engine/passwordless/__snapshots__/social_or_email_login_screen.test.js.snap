// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`email passwordless renders a captcha 1`] = `
<div>
  <div
    data-__type="social_buttons_pane"
    data-instructions="socialLoginInstructions"
    data-labelFn={[Function]}
    data-lock="model"
    data-signUp={true}
  />
  <div
    data-__type="pane_separator"
  />
  <p>
    passwordlessEmailAlternativeInstructions
  </p>
  <div
    data-__type="email_pane"
    data-i18n={
      {
        "html": [Function],
        "str": [Function],
      }
    }
    data-lock="model"
    data-placeholder="emailInputPlaceholder"
    data-strictValidation={false}
  />
  <div
    data-__type="captcha_pane"
    data-flow="passwordless"
    data-i18n={
      {
        "html": [Function],
        "str": [Function],
      }
    }
    data-lock="model"
    data-onReload={[Function]}
  />
</div>
`;

exports[`email passwordless renders correctly 1`] = `
<div>
  <div
    data-__type="social_buttons_pane"
    data-instructions="socialLoginInstructions"
    data-labelFn={[Function]}
    data-lock="model"
    data-signUp={true}
  />
  <div
    data-__type="pane_separator"
  />
  <p>
    passwordlessEmailAlternativeInstructions
  </p>
  <div
    data-__type="email_pane"
    data-i18n={
      {
        "html": [Function],
        "str": [Function],
      }
    }
    data-lock="model"
    data-placeholder="emailInputPlaceholder"
    data-strictValidation={false}
  />
</div>
`;
